Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1080,7 +1080,8 @@ public string GetSmartApiUrl(HttpRequest request)
             if (ConfigurationManager.GetNetworkConfiguration().EnablePublishedServerUriByRequest)
             {
                 int? requestPort = request.Host.Port;
-                if ((requestPort == 80 && string.Equals(request.Scheme, "http", StringComparison.OrdinalIgnoreCase)) || (requestPort == 443 && string.Equals(request.Scheme, "https", StringComparison.OrdinalIgnoreCase)))
+                if (((requestPort == 80 || requestPort == null) && string.Equals(request.Scheme, "http", StringComparison.OrdinalIgnoreCase))
```

Comment:
In theory there could be. But I don't think that can be the case in our current implementation.

But just because it's currently not the case I wouldn't remove the checks

Reply:
There's the case of YAGNI. Would we ever allow ftp, udp etc? Probably not and if we did, this wouldn't be what breaks it.

Reply:
I agree that we could pull out the null check but I'm not sure if we should also drop the schema check for the ports. I'd like to keep them just to make sure that someone doesn't run http on 443 or https on 80 (it could be argued that no sane person would do this, but one never knows...)

Reply:
If they really want to, I don't see why not

Reply:
I pulled out the null check and kept the schema check because if you run https on 80 or http on 443 setting request port to -1 would lead to a wrong URI being created because it falls back to the schema's default port.